dtbase/core/models.py

```diff
@@ -0,0 +1,486 @@
+
+    Args:
+        name: Name of the model measure.
+        session: SQLAlchemy session. Optional.
+
+    Returns:
+        Database id of the model measure.
+    """
+    query = sqla.select(ModelMeasure.id).where(ModelMeasure.name == name)
+    result = session.execute(query).fetchall()
+    if len(result) == 0:
+        raise ValueError(f"No model measure '{name}'.")
+    return result[0][0]
+
+
+@add_default_session
+def model_id_from_name(name, session=None):
+    """Find the id of a model of the given name.
+
+    Args:
+        name: Name of the model.
+        session: SQLAlchemy session. Optional.
+
+    Returns:
+        Database id of the model.
+    """
+    query = sqla.select(Model.id).where(Model.name == name)
+    result = session.execute(query).fetchall()
+    if len(result) == 0:
+        raise ValueError(f"No model named '{name}'")
+    return result[0][0]
+
+
+@add_default_session
+def insert_model(name, session=None):
+    """Insert a new model into the database.
+
+    Args:
+        name: Name of the model.
+        session: SQLAlchemy session. Optional.
+
+    Returns:
+        The new Model
+    """
+    new_model = Model(name=name)
+    session.add(new_model)
+    session.flush()
+    return new_model
+
+
+@add_default_session
+def insert_model_scenario(model_name, description, session=None):
+    """Insert a new model scenario into the database.
+
+    A model scenario specifies parameters for running a model. It is always tied to a
+    particular model. It comes with a free form text description only (can also be
+    null).
+
+    Args:
+        model_name: Name of this model type
+        description: Free form text description. Can also be None/null.
+        session: SQLAlchemy session. Optional.
+
+    Returns:
+        The new ModelScenario
+    """
+    model_id = model_id_from_name(model_name, session=session)
+    new_scenario = ModelScenario(model_id=model_id, description=description)
+    session.add(new_scenario)
+    session.flush()
+    return new_scenario
+
+
+@add_default_session
+def insert_model_measure(name, units, datatype, session=None):
+    """Insert a new model measure into the database.
+
+    Model measures are types of values that models can output in their forecasts. For
+    instance "mean temperature" or "upper bound for electricity consumption".
+
+    Args:
+        name: Name of this measure, e.g. "mean temperature".
+        units: Units in which this measure is specified.
+        datatype: Value type of this model measure. Has to be one of "string",
+            "integer", "float", or "boolean".
+        session: SQLAlchemy session. Optional.
+
+    Returns:
+        The new ModelMeasure
+    """
+    if datatype not in ("string", "integer", "float", "boolean"):
+        raise ValueError(f"Unrecognised data type: {datatype}")
+    new_measure = ModelMeasure(name=name, units=units, datatype=datatype)
+    session.add(new_measure)
+    session.flush()
+    return new_measure
```

Comment:
I think this is fine for now, as we're mostly not using return values for anything, but I think once we're done with adding accessors and API endpoints it could be good to harmonize our strategy for what we return for these "insert_x" functions (in locations, sensors, and models).   We could either return:
* None
* id of newly created object
* newly created object
I think any of these are fine, but we should pick one...